### ISSUE TITLE
[FEATURE]: Move `coverage` into `core`

### ISSUE THREAD

**Issue by vil1** [databrickslabs/remorph#909]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Category of feature request

Transpile

### Problem statement

CI times are longer than the should be. That is partly due to the fact that `remorph-core` is built twice : once during `test-core` and once during `coverage-tests`.

### Proposed Solution

A simple solution would be to move `coverage` into `core`, build a jar only once, and then use that jar to run the coverage tests for each dialect.

**Comment by sundarshankar89** [comment]
@vil1 Isn't this already done?

**Comment by vil1** [comment]
Fixed by #912
